xapian-applications/omega/query.cc

```diff
@@ -1567,6 +1568,10 @@ eval(const string &fmt, const vector<string> &param)
 			hexhash += buf;
 		    }
 		    value = hexhash;
+		    flag = true;
+		}
+		if (flag != true) {
+		    throw Xapian::InvalidOperationError("Hashing algorithm is invalid");
```

Comment:
Don't compare a boolean value against `true` or `false` explicitly like this.  Use `if (flag)` or `if (!flag)` instead.

Though here `flag` seems unnecessary - the `throw` can just be an `else` on the `if` which checks for `md5`, which would be simpler and a bit clearer.

`Xapian::InvalidOperationError` is a xapian-core exception - here you want to throw a `std::string` with the message, and then in omegatest you can check the output to be `Exception: ` followed by that string.

It would be useful to mention the value that wasn't handled; I'd also suggest wording it like similar existing errors, such as:

```
            throw "Unknown function '" + var + "'";
```